feat: add vaton 0.1.0-dev to alire

mhatzl opened the pull request:
Hello,
please add *vaton-0.1.0-dev* to Alire

Thanks!

mhatzl commented:
not sure why ubuntu-latest fails to build. Has there been a change to Ada.Numerics?

My GitHub action on ubuntu-latest worked.
Thanks!

mosteo commented on index/va/vaton/vaton-0.1.0-dev.toml:
```diff
@@ -0,0 +1,37 @@
+spark_unbound = "~0.2.1"
+[[depends-on]]
+# Needed for Ada.Numerics.Big_Numbers
+gnat = "(>=9.3.1 & <2000) | >=2021"
```
One possibility for the numerics failure is that GNAT 10 were to old. You could try bumping the minimum version to 11.

mosteo commented on index/va/vaton/vaton-0.1.0-dev.toml:
```diff
@@ -0,0 +1,37 @@
+[[depends-on]]
+# Needed for Ada.Numerics.Big_Numbers
+gnat = "(>=9.3.1 & <2000) | >=2021"
+gnatprove = "^12.1.1"
```
If gnatprove is used to verify the library, but is not needed by users of the library, the proving should be moved to a nested crate to avoid pushing this heavy dependency into users.

mhatzl replied:
I moved it to the tests sub-crate

mosteo commented on index/va/vaton/vaton-0.1.0-dev.toml:
```diff
@@ -0,0 +1,37 @@
+name = "vaton"
+description = "Verified Ascii To Number conversion written in Ada/SPARK"
+version = "0.1.0-dev"
```
In general, for submissions to the public index, we prefer that pre-release versions (the `-dev` part) are not used unless there's a strong reason. If this was by unawareness, simply remove it.

mhatzl replied:
have removed the "-dev" tag. thanks for the hint

mhatzl commented:
Ok good. most checks are passing. The first one fails, because I added three compile switches to `alire.toml`.
Do I just remove them from the index one?

The Ada2022 switch I added because I use Ada2022 features.
The style-checks I disabled because I do not like the enforced spacing errors, and I did not know how to selectively ignore them.
I also had to change the compiler checks for validation, because an instantiation warning of spark_unbound would prevent the build, but I do not care about warnings from other crates.

Hope this is ok

mosteo commented:
Looks good now. The failing test is because the js TOML parser is somehow behind the times, nothing to worry about.
